checks/frozen_deps.go

```diff
@@ -46,14 +46,102 @@ func FrozenDeps(c *checker.CheckRequest) checker.CheckResult {
 		isPackageManagerLockFilePresent,
 		isGitHubActionsWorkflowPinned,
 		isDockerfilePinned,
+		isDockerfileFreeOfInsecureDownloads,
 	)(c)
 }
 
-// TODO(laurent): need to support GCB
+// TODO(laurent): need to support GCB pinning.
+
+func isDockerfileFreeOfInsecureDownloads(c *checker.CheckRequest) checker.CheckResult {
+	return CheckFilesContent(CheckFrozenDeps, "*Dockerfile*", false, c, validateDockerfileDownloads)
+}
+
+func validateDockerfileDownloads(pathfn string, content []byte,
+	logf func(s string, f ...interface{})) (bool, error) {
+	contentReader := strings.NewReader(string(content))
+	res, err := parser.Parse(contentReader)
+	if err != nil {
+		return false, fmt.Errorf("cannot read dockerfile content: %w", err)
+	}
+
+	ret := true
+	files := make(map[string]bool)
+
+	// Walk the Dockerfile's AST.
+	for _, child := range res.AST.Children {
+		cmdType := child.Value
+		// Only look for the 'RUN' command.
+		if cmdType != "run" {
+			continue
+		}
+
+		var valueList []string
+		for n := child.Next; n != nil; n = n.Next {
+			valueList = append(valueList, n.Value)
+		}
+
+		if len(valueList) != 1 {
```

Comment:
Does this capture both: 

```
RUN bash blah...
```

and 

```
RUN ["/bin/bash", "-c", "..."] 
```

?

Reply:
Thanks. I did not even know this was valid for `RUN` command. Added support.